src/JustEat.StatsD/StatsDImmediatePublisher.cs

```diff
@@ -10,80 +10,88 @@ public class StatsDImmediatePublisher : IStatsDPublisher
     {
         private static readonly CultureInfo SafeDefaultCulture = new CultureInfo(StatsDMessageFormatter.SafeDefaultIsoCultureId);
         private readonly StatsDMessageFormatter _formatter;
-        private readonly IStatsDUdpClient _transport;
+        private readonly IStatsDClient _client;
         private bool _disposed;
 
-        public StatsDImmediatePublisher(CultureInfo cultureInfo, string hostNameOrAddress, int port = 8125, string prefix = "")
+        public StatsDImmediatePublisher(CultureInfo cultureInfo, string hostNameOrAddress, int port = 8125, string prefix = "", Transport transport = Transport.Udp)
         {
             _formatter = new StatsDMessageFormatter(cultureInfo, prefix);
-            _transport = new StatsDUdpClient(hostNameOrAddress, port);
+            if (transport == Transport.Tcp)
```

Comment:
Maybe just inject the implementation instead? Then if we add a 3rd transport this class doesn't have to change.